src/components/layout/Header.tsx

```
@@ -1,88 +1,399 @@
 
 export function Header() {
   const t = useTranslations("nav")
-  const locale = useLocale()
   const router = useRouter()
+  const pathname = usePathname()
   const { data: session } = useSession()
+  const [drawerOpen, setDrawerOpen] = useState(false)
+
+  // Cerrar el drawer al navegar (cuando el path cambia).
+  useEffect(() => {
+    setDrawerOpen(false)
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/layout/Header.tsx:53:5
  51 |   // Cerrar el drawer al navegar (cuando el path cambia).
  52 |   useEffect(() => {
> 53 |     setDrawerOpen(false)
     |     ^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  54 |   }, [pathname])
  55 |
  56 |   const navItems: NavItem[] = [

(react-hooks/set-state-in-effect)